qlty-cli/src/commands/coverage/complete.rs

```diff
@@ -0,0 +1,403 @@
+                "Unknown"
+            }
+        } else {
+            "ERROR"
+        };
+        eprintln_unless!(self.quiet, "    Auth Method: {}", token_type);
+        eprintln_unless!(self.quiet, "    Token: {}", token);
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn print_complete_success(&self, elapsed_seconds: f32) {
+        eprintln_unless!(
+            self.quiet,
+            "    Coverage marked as complete in {:.2}s!",
+            elapsed_seconds
+        );
+        eprintln_unless!(self.quiet, "");
+    }
+
+    fn load_auth_token(&self) -> Result<String> {
+        self.expand_token(match &self.token {
+            Some(token) => Ok(token.to_owned()),
+            None => std::env::var("QLTY_COVERAGE_TOKEN").map_err(|_| {
+                anyhow::Error::msg("QLTY_COVERAGE_TOKEN environment variable is required.")
+            }),
+        }?)
+    }
+
+    fn request_complete(
+        &self,
+        metadata: &qlty_types::tests::v1::CoverageMetadata,
+        token: &str,
+    ) -> Result<Value> {
+        let client = QltyClient::new(Some(LEGACY_API_URL), Some(token.into()));
+        let response_result = client.post("/coverage/complete").send_json(ureq::json!({
+            "data": metadata,
+        }));
+
+        match response_result {
+            Ok(resp) => resp.into_json::<Value>().map_err(|err| {
+                anyhow!(
+                    "JSON Error: {}: Unable to parse JSON response from success: {:?}",
+                    client.base_url,
+                    err
+                )
+            }),
+
+            Err(ureq::Error::Status(code, resp)) => match resp.into_string() {
+                Ok(body) => match serde_json::from_str::<Value>(&body) {
+                    Ok(json) => match json.get("error") {
+                        Some(error) => {
+                            bail!("HTTP Error {}: {}: {}", code, client.base_url, error)
+                        }
+                        None => {
+                            bail!("HTTP Error {}: {}: {}", code, client.base_url, body);
+                        }
+                    },
+                    Err(_) => bail!(
+                        "HTTP Error {}: {}: Unable to parse JSON response: {}",
+                        code,
+                        client.base_url,
+                        body
+                    ),
+                },
+                Err(err) => bail!(
+                    "HTTP Error {}: {}: Error reading response body: {:?}",
+                    code,
+                    client.base_url,
+                    err
+                ),
+            },
+            Err(ureq::Error::Transport(transport_error)) => bail!(
+                "Transport Error: {}: {:?}",
+                client.base_url,
+                transport_error
+            ),
+        }
+    }
+
+    /// Appends repository name to token if it is a workspace token
+    fn expand_token(&self, token: String) -> Result<String> {
```

Comment:
This logic is specifically for dealing with Workspace coverage tokens. With Workspace tokens, we need to work out the repository name via other metadata since the token isn't project specific.

Therefore, this concern feels like it should naturally live in the qlty-coverage crate